frontend/src/pages/user/seller/allCoupons/newCoupon.tsx

```diff
@@ -30,74 +37,168 @@ const tagStyle = {
+        description: '',
+        discount: 0,
+        expire_date: formatDate(new Date().toISOString()),
+        name: '',
+        scope: 'shop',
+        start_date: formatDate(new Date().toISOString()),
+        type: 'percentage',
+      },
       tags: [],
     },
   });
   const { fields, append, remove } = useFieldArray({
     control,
     name: 'tags',
   });
-  const OnFormOutput: SubmitHandler<CouponProps> = (data) => {
-    console.log(data);
-    return data;
-  };
+
   const watchAllFields = watch();
-  const getAllFields = (): CouponProps => {
-    return {
-      id: watchAllFields.id,
-      type: watchAllFields.type,
-      name: watchAllFields.name,
-      description: watchAllFields.description,
-      discount: watchAllFields.discount,
-      start_date: watchAllFields.start_date,
-      expire_date: watchAllFields.expire_date,
-      tags: watchAllFields.tags,
-    };
-  };
 
-  // tags
-  const [tag, setTag] = useState('');
-  const addNewTag = (event: React.KeyboardEvent<HTMLInputElement>) => {
+  const addNewTag = async (event: React.KeyboardEvent<HTMLInputElement>) => {
     // this addressed the magic number: https://github.com/facebook/react/issues/14512
     if (event.keyCode === 229) return;
     if (event.key === 'Enter') {
-      const input = event.currentTarget.value.trim();
-      if (input !== '') {
-        //TODO: api request
-        append({ name: input });
-        setTag('');
+      const newTagName = event.currentTarget.value.trim();
+      if (newTagName !== '') {
+        const foundOldTag = suggestTags.find((tag) => tag.name === newTagName);
+        if (foundOldTag) {
+          // using old tag
+          if (fields.find((tag) => tag.name === newTagName)) {
+            alert('tag already exist');
+            setTag('');
+            return;
+          }
+          append(foundOldTag);
+          setTag('');
+        } else {
+          // creating a new tag
+          const resp = await fetch('/api/seller/tag', {
+            method: 'POST',
+            headers: {
+              accept: 'application/json',
+              'Content-Type': 'application/json',
+            },
+            // TODO: change seller_name to real user name
+            body: JSON.stringify({ name: newTagName, seller_name: 'user1' }),
+          });
+          if (!resp.ok) {
+            alert('error when creating new tag');
+            return;
+          } else {
+            const response = await resp.json();
+            append({ name: newTagName, tag_id: response.id });
+            setTag('');
+          }
+        }
       }
     }
   };
+  const onChangeNewTag = async (event: React.ChangeEvent<HTMLInputElement>) => {
+    setTag(event.target.value);
+    const resp = await fetch(`/api/seller/tag?name=${event.target.value}`, {
+      method: 'GET',
+      headers: {
+        accept: 'application/json',
+      },
+    });
+    if (!resp.ok) {
+      alert('error when fetching existing tag');
```

Comment:
here if i input one letter "i", it will query the existed tag
<img width="436" alt="image" src="https://github.com/jykuo-love-shiritori/twp/assets/92158595/ceb1a08a-8e33-42fc-b116-d728fc86a198">
then i may found that i mistyped, so i delete the letter "i" and want to retype whole word(i haven't do that), it will give me
<img width="733" alt="image" src="https://github.com/jykuo-love-shiritori/twp/assets/92158595/f626c078-3797-43d6-98c3-b70004fdb19e">
this should not be an error because maybe i just want to type new word from start